Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java

```diff
@@ -709,6 +761,86 @@ public boolean hasSearchQuery() {
         return mSearchString != null && !mSearchString.equals("");
     }
 
+    public void addSearchItem(String item, int index) {
+        List<String> recents = new ArrayList<>();
+        Query<Search> query = mBucketSearch.query();
+
+        try (ObjectCursor<Search> cursor = query.execute()) {
+            Search search = mBucketSearch.get(KEY_SEARCH);
+
+            if (cursor.moveToNext()) {
+                recents = cursor.getObject().getRecent();
+            }
+
+            recents.remove(item);
+            recents.add(index, item);
+            // Trim recent searches to MAX_RECENT_SEARCHES (currently 5) if size is greater than MAX_RECENT_SEARCHES.
+            search.setRecent(recents.subList(0, recents.size() > MAX_RECENT_SEARCHES ? MAX_RECENT_SEARCHES : recents.size()));
+            search.save();
+        } catch (BucketObjectMissingException exception) {
+            // Create entity for recent searches since it doesn't exist.
+            try {
+                Search search = mBucketSearch.newObject(KEY_SEARCH);
+                search.save();
+            } catch (BucketObjectNameInvalid invalid) {
+                Log.e("addSearchItem", "Could not create search object", invalid);
+            }
+        }
+    }
```

Comment:
Just for fun and to help re-familiarize myself with Java I have created a version of add/remove using Streams

https://repl.it/repls/ZealousOrchidLinkedlist

```java
public static final int MAX_SEARCHES = 5;

public static List<String> useSearch(List<String> searches, String search) {
  return Stream.concat(
    Stream.of(search),
    searches.stream().filter(s -> !s.equals(search))
  )
    .limit(MAX_SEARCHES)
    .collect(Collectors.toList());
}

public static List<String> removeSearch(List<String> searches, String search) {
  return searches
    .stream()
    .filter(s -> !s.equals(search))
    .collect(Collectors.toList());
}
```